src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -1667,6 +1693,69 @@ protected override void HandleModulesRequestAsync(IRequestResponder<ModulesArgum
             responder.SetResponse(response);
         }
 
+        // test -- need to delete
+        protected override void HandleDisassembleRequestAsync(IRequestResponder<DisassembleArguments, DisassembleResponse> responder)
+        {
+            DisassembleResponse response = new DisassembleResponse();
+            DisassembleArguments disassembleArguments = responder.Arguments;
+            Debug.Assert(!string.IsNullOrEmpty(disassembleArguments.MemoryReference));
+            try
+            {
+                if (GetMemoryContext(disassembleArguments.MemoryReference, disassembleArguments.Offset, out IDebugMemoryContext2 memoryContext, out ulong address) == HRConstants.S_OK)
+                {
+                    IDebugCodeContext2 codeContext = memoryContext as IDebugCodeContext2;
+                    if (!(codeContext is IDebugCodeContext2))
```

Comment:
when will this actually be false? Your line above is the cast. I think your check here should be if `codeContext == null` indicating a cast failure instead of a type check.

Reply:
See latest commit.